internal/store/json.go

```diff
@@ -0,0 +1,303 @@
+	workspace := make(map[string]blueprint.Blueprint)
+	for name, blueprint := range workspaceStruct {
+		workspace[name] = blueprint.DeepCopy()
+	}
+	return workspace
+}
+
+func newComposesFromV0(composesStruct composesV0) map[uuid.UUID]compose.Compose {
+	composes := make(map[uuid.UUID]compose.Compose)
+
+	for composeID, composeStruct := range composesStruct {
+		c := compose.Compose{
+			Blueprint: composeStruct.Blueprint,
+		}
+		if len(composeStruct.ImageBuilds) == 0 {
+			panic("the was a compose with zero image builds, that is forbidden")
+		}
+		for _, imgBuild := range composeStruct.ImageBuilds {
+			ib := compose.ImageBuild{
+				Id:          imgBuild.ID,
+				ImageType:   imgBuild.ImageType,
+				Manifest:    imgBuild.Manifest,
+				Targets:     imgBuild.Targets,
+				JobCreated:  imgBuild.JobCreated,
+				JobStarted:  imgBuild.JobStarted,
+				JobFinished: imgBuild.JobFinished,
+				Size:        imgBuild.Size,
+				JobId:       imgBuild.JobID,
+				QueueStatus: imgBuild.QueueStatus,
+			}
+			c.ImageBuilds = append(c.ImageBuilds, ib)
+		}
+		composes[composeID] = c
+	}
+
+	return composes
+}
+
+func newSourceConfigsFromV0(sourcesStruct sourcesV0) map[string]SourceConfig {
+	sources := make(map[string]SourceConfig)
+
+	for name, source := range sourcesStruct {
+		sources[name] = SourceConfig(source)
+	}
+
+	return sources
+}
+
+func newChangesFromV0(changesStruct changesV0) map[string]map[string]blueprint.Change {
+	changes := make(map[string]map[string]blueprint.Change)
+
+	for name, commitsStruct := range changesStruct {
+		commits := make(map[string]blueprint.Change)
+		for commitID, change := range commitsStruct {
+			commits[commitID] = blueprint.Change{
+				Commit:    change.Commit,
+				Message:   change.Message,
+				Revision:  change.Revision,
+				Timestamp: change.Timestamp,
+			}
+		}
+		changes[name] = commits
+	}
+
+	return changes
+}
+
+func newCommitsFromV0(commitsStruct commitsV0) map[string][]string {
+	commits := make(map[string][]string)
+	for name, changes := range commitsStruct {
+		commits[name] = changes
+	}
+	return commits
+}
+
+func newStoreFromV0(storeStruct storeV0) *Store {
+	store := Store{
+		blueprints:        newBlueprintsFromV0(storeStruct.Blueprints),
+		workspace:         newWorkspaceFromV0(storeStruct.Workspace),
+		composes:          newComposesFromV0(storeStruct.Composes),
+		sources:           newSourceConfigsFromV0(storeStruct.Sources),
+		blueprintsChanges: newChangesFromV0(storeStruct.Changes),
+		blueprintsCommits: newCommitsFromV0(storeStruct.Commits),
+	}
+
+	// Backwards compatibility: fail all builds that are queued or
+	// running. Jobs status is now handled outside of the store
+	// (and the compose). The fields are kept so that previously
+	// succeeded builds still show up correctly.
+	for composeID, compose := range store.composes {
+		if len(compose.ImageBuilds) == 0 {
+			panic("the was a compose with zero image builds, that is forbidden")
```

Comment:
same as above

Reply:
Thanks!

Reply:
Ah, this was dropped in a later patch. I'll leave the typo in the git history ;)